src/ansi-c/c_preprocess.cpp

```diff
@@ -651,192 +555,44 @@ bool c_preprocess_gcc_clang(
-  command+=" -D__FLT_MIN_10_EXP__=\"(-37)\"";
-  command+=" -D__DEC128_MAX_EXP__=6144";
-  command+=" -D__GNUC_MINOR__=2";
-  command+=" -D__DBL_MAX_10_EXP__=308";
-  command+=" -D__LDBL_DENORM_MIN__=3.64519953188247460253e-4951L";
-  command+=" -D__DEC128_MANT_DIG__=34";
-  command+=" -D__LDBL_MIN_10_EXP__=\"(-4931)\"";
+  command +=" -E -D__CPROVER__";
+
+  if(config.ansi_c.pointer_width==16)
+    command+=" -m16";
+  else if(config.ansi_c.pointer_width==32)
+    command+=" -m32";
+  else if(config.ansi_c.pointer_width==64)
+    command+=" -m64";
 
   if(preprocessor==configt::ansi_ct::preprocessort::CLANG)
   {
+    // not sure about the below
     command+=" -D_Noreturn=\"__attribute__((__noreturn__))\"";
-    command+=" -D__llvm__";
-    command+=" -D__clang__";
-  }
-
-  if(config.ansi_c.int_width==16)
-    command+=GCC_DEFINES_16;
-  else if(config.ansi_c.int_width==32)
-  {
-    if(config.ansi_c.pointer_width==64)
-    {
-      if(config.ansi_c.long_int_width==32)
-        command+=GCC_DEFINES_LLP64; // Windows, for instance
-      else
-        command+=GCC_DEFINES_LP64;
-    }
-    else
-      command+=GCC_DEFINES_32;
   }
 
   // The width of wchar_t depends on the OS!
-  {
-    command+=" -D__WCHAR_MAX__="+type_max(wchar_t_type());
-
-    std::string sig=config.ansi_c.wchar_t_is_unsigned?"unsigned":"signed";
-
-    if(config.ansi_c.wchar_t_width==config.ansi_c.short_int_width)
-      command+=" -D__WCHAR_TYPE__=\""+sig+" short int\"";
-    else if(config.ansi_c.wchar_t_width==config.ansi_c.int_width)
-      command+=" -D__WCHAR_TYPE__=\""+sig+" int\"";
-    else if(config.ansi_c.wchar_t_width==config.ansi_c.long_int_width)
-      command+=" -D__WCHAR_TYPE__=\""+sig+" long int\"";
-    else if(config.ansi_c.wchar_t_width==config.ansi_c.char_width)
-      command+=" -D__WCHAR_TYPE__=\""+sig+" char\"";
-    else
-      UNREACHABLE;
-  }
+  if(config.ansi_c.wchar_t_width==config.ansi_c.short_int_width)
+    command+=" -fshort-wchar";
 
   if(config.ansi_c.char_is_unsigned)
-    command+=" -D __CHAR_UNSIGNED__"; // gcc
-
-  switch(config.ansi_c.os)
-  {
-  case configt::ansi_ct::ost::OS_LINUX:
-    command+=" -Dlinux -D__linux -D__linux__ -D__gnu_linux__";
-    command+=" -Dunix -D__unix -D__unix__";
-    command+=" -D__USE_UNIX98";
-    break;
-
-  case configt::ansi_ct::ost::OS_MACOS:
-    command+=" -D__APPLE__ -D__MACH__";
-    // needs to be __APPLE_CPP__ for C++
-    command+=" -D__APPLE_CC__";
-    break;
-
-  case configt::ansi_ct::ost::OS_WIN:
-    command+=" -D _WIN32";
-
-    if(config.ansi_c.mode!=configt::ansi_ct::flavourt::VISUAL_STUDIO)
-      command+=" -D _M_IX86=Blend";
-
-    if(config.ansi_c.arch=="x86_64")
-      command+=" -D _WIN64"; // yes, both _WIN32 and _WIN64 get defined
-
-    if(config.ansi_c.char_is_unsigned)
-      command+=" -D _CHAR_UNSIGNED"; // This is Visual Studio
-    break;
-
-  case configt::ansi_ct::ost::NO_OS:
-    command+=" -nostdinc"; // make sure we don't mess with the system library
```

Comment:
This should still be handled/used.

Reply:
Not sure I want to retain like this, but ok for now.